core/src/main/kotlin/org/evomaster/core/database/SqlInsertBuilder.kt

```diff
@@ -638,6 +638,26 @@ class SqlInsertBuilder(
         }
     }
 
+    /**
+     * extract tables with additional fk tables
+     * @param tables to check
+     * @param all is a complete set of tables with their fk
+     */
+    fun extractFkTable(tables: Set<String>, all: MutableSet<String>){
+        tables.forEach { t->
```

Comment:
i guess here we should return the `all` set, as immutable `Set`. also, to deal with recursion, the input `all` could have a default `= mutableSet<String>()`

Reply:
updated